src/main/java/de/flo56958/minetinker/listeners/AnvilListener.java

```diff
@@ -153,50 +157,63 @@ public void onAnvilPrepare(@NotNull final PrepareAnvilEvent event) {
 			return;
 		}
 
-		if (!(modManager.isToolViable(tool) || modManager.isArmorViable(tool))) {
+		if (!(modManager.isToolViable(item1) || modManager.isArmorViable(item1))) {
 			return;
 		}
 
-		if (modifier.getType().equals(Material.ENCHANTED_BOOK)) { //So no Tools can be enchanted via books, if enchanting is disabled
-			if (MineTinker.getPlugin().getConfig().getBoolean("AllowEnchanting")) {
-				// If enchanting is allowed, don't do anything
-				return;
-			} else {
-				// Otherwise, set the resulting item to AIR to negate the enchant
+		if (item2.getType().equals(Material.ENCHANTED_BOOK)) { //So no Tools can be enchanted via books, if enchanting is disabled
+			if (!MineTinker.getPlugin().getConfig().getBoolean("AllowEnchanting")) {
+				//Set the resulting item to AIR to negate the enchant
 				event.setResult(new ItemStack(Material.AIR, 0)); //sets ghostitem by client
-				return;
 			}
+			return;
 		}
 
-		final Modifier mod = modManager.getModifierFromItem(modifier);
+		final Modifier mod = modManager.getModifierFromItem(item2);
 
 		ItemStack newTool = null;
 
 		if (mod != null) {
-			newTool = tool.clone();
-			if (!modManager.addMod(player, newTool, mod, false, false, false)) {
+			newTool = item1.clone();
+			if (!modManager.addMod(player, newTool, mod, false, false, false, true)) {
 				return;
 			}
+		} else if (item1.getType() == item2.getType()) { //Whether we're combining the tools
+			if (MineTinker.getPlugin().getConfig().getBoolean("Combinable")
+					&& player.hasPermission("minetinker.tool.combine")) {
+				newTool = item1.clone();
+
+				for (Modifier tool2Mod : modManager.getToolMods(item2)) {
+					int modLevel = modManager.getModLevel(item2, tool2Mod);
+					for (int i = 0; i < modLevel; i++) {
+						if (!modManager.addMod(player, newTool, tool2Mod, false, false, true, false)) {
+							return;
```

Comment:
I think you want to continue the outer loop and not return. If you continue the Modifier just gets discarded and does not break the feature entirely.